compiler.pri

```diff
@@ -227,14 +227,44 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# Allow CONFIG(c++11) to enable C++11 mode
+	# when building against Qt 4.
+	lessThan(QT_MAJOR_VERSION, 5):CONFIG(c++11) {
```

Comment:
No.

QT_CONFIG is populated when Qt is built.
Internally in Qt's build system, the c++11, c++14, c++1z are available via CONFIG(c++11), etc. -- but not in third party code. (It's in an internal qmake 'feature'/prf file, qt_common.prf, I believe).
For third party code, as far as I can tell, the only way to detect whether Qt was built in c++11/14/1z mode, is to check QT_CONFIG.

The CONFIG(c++11) option for Qt 4 is meant as a band-aid for distros that only ship C++11 variants of some of our dependencies. In that case, they can simply add CONFIG+=c++11 to their build recipe, instead of having to patch Mumble.
It can be seen as something separate from the other auto-detection logic. Qt 4 doesn't have any C++11 features, so the builds are identical as far as Qt is concerned.
The reason the option was added was because of https://github.com/mumble-voip/mumble/issues/2494.